install.sh

```diff
@@ -463,6 +463,51 @@ assert_usable_kernel() {
   fi
 }
 
+maybe_enable_userns_sysctl() {
+  # This function enables the Debian/Ubuntu-specific unprivileged
+  # userns sysctl, if the system has it and we want it.
+
+  if [ "$USE_DEFAULTS" != "yes" ] ; then
+    # Only do this when -d is passed. -d means "use defaults suitable for app development", and
+    # we want userns enabled for app development if possible since it enables UID randomization
+    # which helps catch app bugs. For the rest of the world, we're fine using the privileged
+    # sandbox instead.
+    return
+  fi
+
+  if [ "no" = "$CURRENTLY_UID_ZERO" ] ; then
+    # Not root. Can't do anything about it.
+    return
+  fi
+
+  if [ ! -e /proc/sys/kernel/unprivileged_userns_clone ]; then
+    # No such sysctl on this system.
+    return
+  fi
+
+  if [ "$(sysctl -n kernel.unprivileged_userns_clone)" = "1" ]; then
+    # Already enabled.
+    return
+  fi
+
+  # Enable it.
```

Comment:
If `/proc/sys` is read-only, we should probably avoid setting the flag. It's read-only within Docker.

We have the following code currently within install.sh:

```
  # The /proc/sys issue applies to Docker containers and probably other sorts of containers, too.
  if egrep -q '/proc/sys\s+proc\s+ro' /proc/mounts || [ ! -e /etc/sysctl.conf ] ; then
```

Presumably we should use the same `egrep` test.